src/FluentValidation/Internal/MessageFormatter.cs

```diff
@@ -93,21 +92,26 @@ public MessageFormatter AppendPropertyValue(object value)
 		/// </summary>
 		public Dictionary<string, object> PlaceholderValues => _placeholderValues;
 
-		protected virtual string ReplacePlaceholderWithValue(string template, string key, object value) {
-			string placeholder =  GetPlaceholder(key);
-			return template.Replace(placeholder, value?.ToString());
-		}
+		// TODO: Add alignment component
+		private Regex _keyRegex = new Regex("{([^{}:]+)(?::([^{}]+))?}"); // Matches any placeholder
 
-		protected string GetPlaceholder(string key) {
-			// Performance: String concat causes much overhead when not needed. Concatting constants results in constants being compiled.
-			switch (key) {
-				case PropertyName:
-					return "{" + PropertyName + "}";
-				case PropertyValue:
-					return "{" + PropertyValue + "}";
-				default:
-					return "{" + key + "}";
-			}
+		protected virtual string ReplacePlaceholdersWithValues(string template, IDictionary<string, object> values)
+		{

```

Comment:
I have 2 concerns with this:

1) This is a breaking change for anyone that inherits from `MessageFormatter` (so this would have to be a major release and the next one isn't scheduled until next year)

2) Use of a regex for token replacement is likely to have a performance impact on high-traffic systems. 

The previous `GetPlaceholder` was high performance to account for this (this is one of the codebase's hot-spots). Would it be possible to change the approach here to only use the regex in the situation where formatted placeholders are in use? If the message doesn't have any, then fall back to the existing high-performance behaviour (which also preserves backwards compatibility for inheritors).

One approach could be:
- `BuildMessage` should detect if the `messageTemplate` contains formatted placeholders
- If not, it uses the original logic (loop over placeHolder values and call `ReplacePlaceholderWithValue`)
- If it does, then it uses the new logic. 

Thoughts?